docker/Dockerfile

```diff
@@ -56,17 +56,17 @@ ARG VERSION="unknown"
 ARG GIT_COMMIT="unknown"
 RUN make cmds
 
-FROM ${CUDA_SAMPLE_IMAGE} AS sample-builder
-
-FROM nvcr.io/nvidia/cuda:12.9.0-base-ubi9
+# Download must-gather dependency: `kubectl`
+ARG TARGETARCH
+RUN OS_ARCH=${TARGETARCH/x86_64/amd64} && OS_ARCH=${OS_ARCH/aarch64/arm64} && \
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/${OS_ARCH}/kubectl && \
+    chmod +x ./kubectl
 
-# Remove CUDA libs(compat etc) in favor of libs installed by the NVIDIA driver
-RUN dnf remove -y cuda-*
+FROM ${CUDA_SAMPLE_IMAGE} AS sample-builder
 
-RUN dnf install -y \
-        kmod \
-        pciutils && \
-    rm -rf /var/cache/yum/*
+# The C/C++ distroless image is used as a base since the CUDA vectorAdd
+# sample application depends on C/C++ libraries.
```

Comment:
Not in scope for this PR, but does this mean that we also use this container to validate functionality? Why do we reference the CUDA samples images in other places then?

Reply:
This is a fair question that definitely warrants discussion. I would need to sync up with @shivamerla to better understand the motivation behind the current CUDA validation workflow. This is out of scope for this PR.

As I currently understand it, we do not actually deploy the CUDA samples image at runtime -- we only use it at build time to extract the `vectorAdd` binary and include it in the validator image. At runtime, we perform "CUDA validation" by executing the `vectorAdd` sample that is included in the validator image.

Reply:
OK. That is good to know. I suppose that's why the sample images have not really been kept up to date.

One thing to be aware of here is whether CUDA Forward Compatibility might then become an issue for this image -- especially since we are removing the CUDA Forward Compatibility libraries from this image by switching to distroless.

Reply:
Thanks for mentioning CUDA Forward Compatibility. Without the compat libraries, the samples would fail to run when the major version of the CUDA driver installed is **_less than_** the major version that the CUDA sample is built against, correct?

What about including the CUDA compat libs in the final image like below?
```
FROM nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda12.5.0 AS sample-builder

FROM nvcr.io/nvidia/distroless/cc:v3.1.7-dev

COPY --from=sample-builder /usr/local/cuda/compat /usr/local/cuda/compat
```

Reply:
"Copying the cuda compat libraries" is a reasonable approach to me. As I understand it, we need it so our current validation workflows don't break. Given that the validator image runs the `vectorAdd` binary to perform validation, it makes sense to add it here as we are no longer using a separate "operator-validator" image.

Reply:
I went ahead and made this change. However, the [image builds for ARM](https://github.com/NVIDIA/gpu-operator/actions/runs/15935880646/job/44955747604?pr=1417#step:6:276) are failing with:
```
>>> COPY --from=sample-builder /usr/local/cuda/compat /usr/local/cuda/compat
  .
  .
  .
"/usr/local/cuda/compat": not found
```
I inspected the CUDA base images for ARM, and it looks like the compat libs are only included starting with 12.5.1. The base images I inspected that are <= 12.5.0 don't have the compat libs.

@elezar @tariq1890 what are your thoughts on if we published new CUDA samples images for the latest version of CUDA (or a more recent version)? That would unblock this PR and allow us to include the compat libs in the operator image.